test/tests/functional/pbs_multi_sched.py

```diff
@@ -699,3 +699,128 @@ def test_cmp_fairshare(self):
 
         user = self.scheds['sc3'].cmp_fairshare_entities(TEST_USER, TEST_USER2)
         self.assertEquals(user, str(TEST_USER))
+
+    def test_pbfs_invalid_sched(self):
+        """
+        Test pbsfs -I <sched_name> where sched_name does not exist
+        """
+        sched_name = 'foo'
+        pbsfs_cmd = os.path.join(self.server.pbs_conf['PBS_EXEC'],
+                                 'sbin', 'pbsfs') + ' -I ' + sched_name
+        ret = self.du.run_cmd(cmd=pbsfs_cmd, sudo=True)
+        err_msg = 'Scheduler %s does not exist' % sched_name
+        self.assertEquals(err_msg, ret['err'][0])
+
+    def test_pbfs_no_fairshare_data(self):
+        """
+        Test pbsfs -I <sched_name> where sched_priv_<sched_name> dirs are empty
+        """
+        a = {'partition': 'P5',
+             'sched_host': self.server.hostname,
+             'sched_port': '15050'}
+        self.server.manager(MGR_CMD_CREATE, SCHED, id="sc5")
+        sched_name = 'sc5'
+        err_msg = 'Unable to access fairshare data: No such file or directory'
+
+        try:
+            ret = self.scheds['sc5'].query_fairshare()
+        except PbsFairshareError as e:
+            self.assertTrue(err_msg in e.msg)
+
+    def test_pbfs_server_down(self):
+        """
+        Test pbsfs -I <sched_name> when server is down
+        """
+        self.setup_sc1()
+        err_msg = "Can't connect to the server"
+        try:
+            self.server.stop()
+            n = self.scheds['sc1'].query_fairshare()
+        except PbsFairshareError as e:
+            self.assertTrue(err_msg in e.msg)
+        self.server.start()
+
+    def test_pbfs_server_restart(self):
+        """
+        Test pbsfs -I <sched_name> when server is restarted
+        """
+        self.setup_sc1()
+        self.scheds['sc1'].add_to_resource_group(TEST_USER, 20, 'root', 50)
+        self.scheds['sc1'].set_fairshare_usage(name=TEST_USER, usage=25)
+        n = self.scheds['sc1'].query_fairshare().get_node(name=str(TEST_USER))
+        self.assertTrue(n.usage, 25)
+
+        self.server.restart()
+        n = self.scheds['sc1'].query_fairshare().get_node(name=str(TEST_USER))
+        self.assertTrue(n.usage, 25)
+
+    def test_trim_fairshare(self):
+        """
+        Test pbsfs's trim fairshare tree functionality for multisched
+        """
+        self.common_setup()
+        sched_name = 'sc1'
```

Comment:
Does setup for test needs all scheduers ? else you can just setup_sc1 and queues and nodes as common_setup is heavy . 
Is there any intension of using sched_name variable than just using 'sc1' at all places ?

Reply:
@visheshh . No. There wasn't any specific reason. I have reverted to 'sc1' as commonly done in other tests of the file.  Also, creating 'sc1' specific sched, queue and node. Got rid of common_setup(). Thanks!